packages/bookshelf-include-count/lib/bookshelf-include-count.js

```diff
@@ -157,8 +226,71 @@ module.exports = function (Bookshelf) {
             });
 
             return attrs;
-        }
+        },
+
+        /**
+         * Instead of adding the counts in .fetch and .fetchAll,
+         * we need to do it in sync because Bookshelf doesn't call fetch for eagerRelations
+         * E.g. when trying to load counts on replies.count.likes, we wouldn't get an opportunity to load the counts on the replies relation.
+         */
+        sync: function (options) {
+            if (!options.method || (options.method !== 'insert' && options.method !== 'update')) {
+                this.addCounts.apply(this, arguments);
+            }
+
+            if (_debug.enabled('ghost-query')) {
+                debug('QUERY', this.query().toQuery());
+            }
+
+            // Call parent fetchAll
+            return modelProto.sync.apply(this, arguments);
+        },
+
+        // When save a modal, make sure we keep the already fetched counts alive because they will get
+        // removed before saving in the saving event (as they are not 'permitted')
+        // We need to use bluebird to override save (or things will break)
```

Comment:
Why do things break? `modelProto.save.apply` _should_ return an instance of a Bluebird Promise, what am I missing here?

Reply:
Good remark! This was a leftover from earlier tries. Initially I was using `await` inside `save()` and marked it as an async function, which converted the Bluebird promise to a native Promise, causing it to break (because Bookshelf itself is returning a Bluebird promise and a couple of places depend on that). This is no longer needed.

I'll remove the comment, add a warning to make sure it should stay a bluebird promise and remove the Promise.method wrapping.